go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -238,32 +255,29 @@ func (c *Controller) syncServices(key string) error {
 	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-	// Get current state of the Service from the Service tracker
-	// These are the VIPs (ClusterIP:Port) that we have seen so far
-	// If the Service has updated the VIPs (has changed the Ports)
-	// and some were removed we have to delete those
-	// We need to create a map to not mutate the service tracker VIPs
-	vipsTracked := sets.NewString().Union(c.serviceTracker.getService(name, namespace))
+
 	// Delete the Service VIPs from OVN if:
 	// - the Service was deleted from the cache (doesn't exist in Kubernetes anymore)
 	// - the Service mutated to a new service Type that we don't handle (ExternalName, Headless)
-	if err != nil || !util.ServiceTypeHasClusterIP(service) || !util.IsClusterIPSet(service) {
-		err = deleteVIPsFromAllOVNBalancers(vipsTracked, name, namespace)
-		if err != nil {
-			// If the service wasn't found, don't panic sending an
-			// an event after cleaning it up
-			if service != nil {
-				c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToDeleteOVNLoadBalancer",
-					"Error trying to delete the OVN LoadBalancer for Service %s/%s: %v", name, namespace, err)
-			}
-			return err
+	if err != nil || service == nil || !util.ServiceTypeHasClusterIP(service) || !util.IsClusterIPSet(service) {
+		service = &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      name,
+			},
+		}
+
+		if err := ovnlb.EnsureLBs(util.ExternalIDsForObject(service), nil); err != nil {
+			return fmt.Errorf("failed to delete load balancers for service %s/%s: %w",
+				namespace, name, err)
 		}
-		// Delete the Service form the Service Tracker
-		c.serviceTracker.deleteService(name, namespace)
+
+		c.repair.serviceSynced(key)
 		return nil
 	}
-	klog.Infof("Creating service %s on namespace %s on OVN", name, namespace)
+
 	// The Service exists in the cache: update it in OVN
+
```

Comment:
nit: rm space

Reply:
It's kind of a larger heading. Trying to be vaguely structured. Anyways, tweaked it a bit.